core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,121 @@
+      return new ExplorationOpportunitySummary(opportunityDict.id,
+        opportunityDict.topic_name, opportunityDict.story_title,
+        opportunityDict.chapter_title, opportunityDict.content_count,
+        opportunityDict.translation_counts);
+    } else if (opportunityType === constants.OPPORTUNITY_TYPE_SKILL) {
+      return new SkillOpportunity(
+        opportunityDict.id, opportunityDict.skill_description,
+        opportunityDict.topic_name, opportunityDict.question_count);
+    }
+  }
+
+  // TODO(#7165): Replace any with exact type.
+  private _fetchOpportunities(
+      opportunityType: OpportunityType,
+      params: ContributionOpportunityParams,
+      successCallback: (
+        opportunities?: any, nextCursor?: string, more?: any
+        ) => void,
+      errorCallback: (reason?: any) => void
+  ): void {
+    this.http.get(this.urlInterpolationService.interpolateUrl(
+      this.urlTemplate, { opportunityType }
+    ), { params }).toPromise().then((data: any) => {
+      const opportunities = [];
+      for (const index in data.opportunities) {
+        opportunities.push(this._getOpportunityFromDict(
+          opportunityType, data.opportunities[index]));
       }
-    };
-    var _fetchOpportunities = function(
-        opportunityType, params, successCallback) {
-      return $http.get(
-        UrlInterpolationService.interpolateUrl(
-          urlTemplate, {opportunityType: opportunityType}
-        ), {
-          params: params
-        }).then(function(response) {
-        var data = response.data;
-        var opportunities = [];
-        for (var index in data.opportunities) {
-          opportunities.push(_getOpportunityFromDict(
-            opportunityType, data.opportunities[index]));
-        }
+      if (successCallback) {
         successCallback(opportunities, data.next_cursor, data.more);
-      });
-    };
-    return {
-      fetchSkillOpportunities: function(cursor, successCallback) {
-        var params = {
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_SKILL, params, successCallback);
-      },
-      fetchTranslationOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_TRANSLATION, params, successCallback);
-      },
-      fetchVoiceoverOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_VOICEOVER, params, successCallback);
       }
+    }, (error) => {
+      if (errorCallback) {
+        errorCallback(error);
+      }
+    });
+  }
+
+  fetchSkillOpportunities(
+      cursor: string): Promise<Object> {
```

Comment:
Can we move this to the line above (Following 80 chars in a line)

Reply:
@srijanreddy98 is this resolved?

Reply:
Yes